pkg/http_endpoint/v1/router.go

```diff
@@ -56,26 +60,645 @@ func authRouter(clients *common.ClientContainer,
+                // datasource that doesn't exist. We have to return 404 error then
+                case common.DatasourceNotFound:
+                        common.WriteErrorToResponse(w, 404,
+                                "Datasource not exist", err.Error())
+                        return  
+                // If any other error happened -> return 500 error
+                default:
+                        log.Logger.Error(err)
+                        common.WriteErrorToResponse(w, http.StatusInternalServerError,
+                                http.StatusText(http.StatusInternalServerError),
+                                "Internal server error occured")
+                        return
+                }		
+	}
+	datasources, _ := json.Marshal(dslist)
+	w.WriteHeader(http.StatusOK)
+	w.Write(datasources)
+		}
+}
+
+// DeleteDatasource deletes the datasource by ID
+func DeleteDatasource(clients *common.ClientContainer) http.HandlerFunc {
+                return func(w http.ResponseWriter, r *http.Request) {
+	err := clients.Grafana.DoLogon()
+        if err != nil {
+                switch err.(type) {
+                // common.InvalidUsernamePassword means, that user provided invalid
+                // username/password. We have to return 401 error then
+                case common.InvalidUsernamePassword:
+                        common.WriteErrorToResponse(w, http.StatusUnauthorized,
+                                "Unauthorized", err.Error())
+                        return
+                // If any other error happened -> return 500 error
+                default:
+                        log.Logger.Error(err)
+                        common.WriteErrorToResponse(w, http.StatusInternalServerError,
+                                http.StatusText(http.StatusInternalServerError),
+                                "Internal server error occured")
+                        return
+                }       
+        }
+	datasourceID := chi.URLParam(r, "datasourceID")
+	ID, _ := strconv.Atoi(datasourceID)
+	err = clients.Grafana.DeleteDataSource(ID)
+	if err != nil {
+		log.Logger.Error(err)
+		common.WriteErrorToResponse(w, http.StatusInternalServerError,
+			http.StatusText(http.StatusInternalServerError),
+			"Internal server error occured")
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte(fmt.Sprintf("Sucessfully deleted datasource by ID: %v", datasourceID)))
+		}
+}
+
+//CreateDatasource method is to create  datasource
+func CreateDatasource(clients *common.ClientContainer) http.HandlerFunc {
+		return func(w http.ResponseWriter, r *http.Request) {
+	err := clients.Grafana.DoLogon()
+        if err != nil {
+                switch err.(type) {
+                // common.InvalidUsernamePassword means, that user provided invalid
+                // username/password. We have to return 401 error then
+                case common.InvalidUsernamePassword:
```

Comment:
I don't think it is valid, because it is not User's problem, that our admin grafana creds are invalid.
also, you return err.Error() that can contain some creds.
I would recommend to return 500